command-line/lib/Command/Address/Inspect.hs

```diff
@@ -64,15 +73,26 @@ mod liftCmd = command "inspect" $
   where
     parser = pure Inspect
 
+
+-- used for 'inspect'
+instance Error SomeException
+
 run :: Cmd -> IO ()
 run Inspect = do
     bytes <- hGetBytes stdin
-    case inspect (unsafeMkAddress bytes) of
-        Nothing   -> fail "Unrecognized address on standard input"
-        Just json -> BL8.hPutStrLn stdout (Json.encodePretty json)
+    case first show $ inspect (unsafeMkAddress bytes) of
+      Right json -> BL8.hPutStrLn stdout (Json.encodePretty json)
+      Left  str  -> fail str
```

Comment:
We shouldn't use `Show` instances for user-facing outputs. It'd be better that the errors are rendered properly using a human-readable formatting. `Show` is good for debugging, but not so much for throwing back in users' face ^^" 

There are many pretty-printing libraries available, we use [`fmt`](https://hackage.haskell.org/package/fmt-0.6.1.1/docs/Fmt.html) in `cardano-wallet` (not sure if I like it more than any other but it does the job). So suggestions are welcome too here but we definitely need something that captures the "human readable" aspect of errors that is different from debug Show.

Reply:
I don't have positive experience with pretty printing libraries, they're often overdesigned and awkward to use. I like to use quasiquoters for this kind of stuff, but they will break your tags files :p 

Not really a strong opinion here either.

Reply:
So let's just go for `fmt`, the API is simple and the dependencies reasonable.

Reply:
Note on fmt: it uses OverloadedStrings afais, which is problematic for ByteString (uses Char8). So unicode will not show up correctly. But that's not a problem with String/Text.

Reply:
This turns out to be a little more complex with MonadThrow, since it degrades to `SomeException`, which only has `Show`. We cannot recover it, unless we are in IO and catch for those specific exception types, afais. I think it's fine to abuse Show for this, but ymmv. This was initially done for debugging, which Show is for. Ofc it doesn't hold the `read . show == id` property, but the haskell report only demands that this holds for derived instances afais.